### ISSUE TITLE
current_time in [replace_schedule] not working

### ISSUE THREAD

**Issue by sigurdfdragon** [wesnoth/wesnoth#5757]
### Game and System Information
 - 1.15.12+dev (1cb2393b422-Modified)
 - built from source
 - MSYS method (https://forums.wesnoth.org/viewtopic.php?f=5&t=48296)
 - Win 10

### Describe the bug
The code below switches the schedule, but does not change current time of day to the specified value. The current_time is always the first value in the schedule.
`        [replace_schedule]
            {DEFAULT_SCHEDULE_24H}
            current_time=13
        [/replace_schedule]
`

### To Reproduce
Put the above code in a start event

**Comment by soliton-** [comment]
Was presumably accidentally removed in https://github.com/wesnoth/wesnoth/commit/ebbecb3b047a38cd82c0f01068d982095fc215ed.

**Comment by kallaballa** [comment]
That's quite a big commit. With a little guidance i might be able to fix this. Can someone point me in the right direction?

**Comment by kallaballa** [comment]
found the problem. this line should have been retained: https://github.com/wesnoth/wesnoth/commit/ebbecb3b047a38cd82c0f01068d982095fc215ed#diff-66c87fcf7f50465de685e28a58bc9442ad795af1bfceaff8513dabf7bae63031L305

**Comment by kallaballa** [comment]
Fixing it in my fork

**Comment by kallaballa** [comment]
I fixed it (
ea2e039) and tested it with following code:
```
  [set_variable]
      name=rand_time
      rand=1..6
  [/set_variable]
  [replace_schedule]
      {FIRST_WATCH}
      {SECOND_WATCH}
      {DAWN}
      {MORNING}
      {AFTERNOON}
      {DUSK}
      current_time=$rand_time
  [/replace_schedule]
```

**Comment by kallaballa** [comment]
What would I have to do to fix this in 1.16 as well?
